webapp/src/Entity/Contest.php

```diff
@@ -1260,6 +1260,17 @@ public function getFreezeData(): FreezeData
         return new FreezeData($this);
     }
 
+    public function checkValidTimeString(string $timeString): bool {
+        if (preg_match("/^[+-]?\d+:\d\d(:\d\d(\.\d{1,6})?)?$/", $timeString) !== 1) {
+            try {
+                $date = new DateTime($timeString);
+            } catch (Exception) {
```

Comment:
does the exception here tell us what's wrong? If so, can we extract it (or get it from somewhere else) and return it to the user?